presenter.py

```diff
@@ -0,0 +1,112 @@
+"""
+This module contains Presenter thread class - mediator that contains main game loop
+"""
+from copy import deepcopy
+
+from PySide6 import QtCore
+
+from server import Connection
+from model import GameState, GameMap, GameActions
+from vehicle import Vehicle
+from config.actions import Actions
+
+
+class Presenter(QtCore.QThread):
+    """
+    Mediator class that contains main game loop,
+    runs as a thread called from main window
+    """
+    game_state_updated = QtCore.Signal(object, object)
+    game_ended = QtCore.Signal(str)
+
+    def __init__(self, login_data: dict):
+        super().__init__(None)
+        self.idx = None
+        self.login_data = login_data
+        self.connection = Connection()
+        self.vehicles_list = []
+        self.game_state = None
+        self.game_actions = None
+        self.map = None
+
+    def run(self) -> None:
+        """
+        inherited from QTread method that begins execute when thread started
+        executes game initialisation, runs main game loop, emits signals
+        used to update main window
+        :return: None
+        """
+        turn = None
+        winner = None
+        self.init_game()
+# <---------------------- main loop ---------------------
+        while True:
+            current_game_state = self.connection.send(Actions.GAME_STATE)
```

Comment:
Presenter is supposed to be dealing only with GUI work but it does much more. I works with connections (which are entities of quite low level of abstraction while GUI itself is at a high level of abstraction) and performs some logic like detecting the winner. I would recommend to move all game logic to classes which are currently in the model.py module and leave only GUI-related logic here.